achievements.lua

```diff
@@ -77,37 +77,81 @@
 --]]
 
-local local_achievement_file = "Achievements.json"
+-- TODO?: Maybe make importing these 'extra' graphics optional
+import "CoreLibs/graphics"
 
 -- Right, we're gonna make this easier to change in the future.
 -- Another note: changing the data directory to `/Shared/gameID`
 --   rather than the previously penciled in `/Shared/Achievements/gameID`
+local default_shared_achievement_folder <const> = "/Shared/Data/"
+local default_achievement_file_name <const> = "Achievements.json"
+local default_shared_images_subfolder <const> = "AchievementImages/"
+local default_shared_images_updated_file <const> = "_last_seen_version.txt"
+
+local function basename(str)
+	local pos = str:reverse():find("/", 0, true)
+	if pos == nil then
+		return str
+	end
+	if pos == #str then
+		pos = str:reverse():find("/", 2, true)
+	end
+	return str:sub((#str + 1) - (pos - 1))
+end
+
 local function get_achievement_folder_root_path(gameID)
 	if type(gameID) ~= "string" then
 		error("bad argument #1: expected string, got " .. type(gameID), 2)
 	end
-	local root = string.format("/Shared/%s/", gameID)
+	local root = string.format(default_shared_achievement_folder .. "%s/", gameID)
 	return root
 end
 local function get_achievement_data_file_path(gameID)
 	if type(gameID) ~= "string" then
 		error("bad argument #1: expected string, got " .. type(gameID), 2)
 	end
 	local root = get_achievement_folder_root_path(gameID)
-	return root .. "Achievements.json"
+	return root .. default_achievement_file_name
+end
+local function get_shared_images_path(gameID)
+	if type(gameID) ~= "string" then
+		error("bad argument #1: expected string, got " .. type(gameID), 2)
+	end
+	local root = get_achievement_folder_root_path(gameID)
+	return root .. default_shared_images_subfolder
+end
+local function get_shared_images_updated_file_path(gameID)
+	if type(gameID) ~= "string" then
+		error("bad argument #1: expected string, got " .. type(gameID), 2)
+	end
+	local folder = get_shared_images_path(gameID)
+	return folder .. default_shared_images_updated_file
 end
-
--- local achievement_folder = root_folder .. playdate.metadata.bundleID .. "/"
 
 local metadata <const> = playdate.metadata
+local gfx <const> = playdate.graphics
 
 ---@diagnostic disable-next-line: lowercase-global
 achievements = {
-	version = "prototype 0.1"
+	specversion = "0.1+prototype",
+	libversion = "0.2-alpha+prototype",
+
+	onUnconfigured = error,
+	forceSaveOnGrantOrRevoke = false,
+
+	displayGrantedMilliseconds = 2000,
+	displayGrantedDefaultX = 20,
+	displayGrantedDefaultY = 0,
+	displayGrantedDelayNext = 400,
+	iconWidth = 32,
+	iconHeight = 32,
 }
 
-local function load_granted_data()
-	local data = json.decodeFile(local_achievement_file)
+local function load_granted_data(from_file)
+	if from_file == nil then
+		from_file = "./" .. default_achievement_file_name
+	end
+	local data = json.decodeFile(from_file)
```

Comment:
There's not much point in configuring this to use a variable filepath if the end user can't configure the file used. As things stand, this function is always called with the same argument.

Reply:
Done!